File: backend/src/main/java/com/petrovic/fooddeliveryapi/models/Customer.java
package com.petrovic.fooddeliveryapi.models;

import lombok.*;

import javax.persistence.*;
import java.util.Set;
import java.util.UUID;

@Entity(name = "customers")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Customer {
    @Id
    private String id;
    private String firstName;
    private String lastName;
    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private AppUser appUser;
    @OneToMany(mappedBy = "customer")
    private Set<ShoppingCart> shoppingCarts;

    @PrePersist
    public void prePersist() {
        if (id == null) {
            id = UUID.randomUUID().toString();
        }
    }
}

File: frontend/src/models/Customer.ts
import {AppUser} from "./user";

export class Customer {
    id: number;
    firstName: string;
    lastName: string;
    appUser: AppUser;

    constructor(
        id: number,
        firstName: string,
        lastName: string,
        appUser: AppUser
    ) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.appUser = appUser;
    }
}